homeassistant/components/harmony/data.py

```diff
@@ -35,16 +35,26 @@ def __init__(self, hass, address: str, name: str, unique_id: str):
         )
 
     @property
-    def activity_names(self):
-        """Names of all the remotes activities."""
+    def activities(self):
+        """List of activity_id: activity_name pairs for all the remotes activities."""
+        activity_list = []
+
         activity_infos = self._client.config.get("activity", [])
-        activities = [activity["label"] for activity in activity_infos]
+        for info in activity_infos:
+            activity_id = info["id"]
+            name = info["label"]
+
+            # Filter both ways of representing PowerOff
+            if name is not None and name != ACTIVITY_POWER_OFF:
+                activity_list.append({"id": activity_id, "name": name})
 
```

Comment:
```suggestion
        return [
            info for info in activity_infos if info["name"] is not None and info["id"] != ACTIVITY_POWER_OFF_ID
        ]
```

Assumes `ACTIVITY_POWER_OFF_ID` is always `-1`


Can we keep the data the same and use `label` in `__init__`.

Reply:
Needed to rejigger this a bit (needed to compare using `info["label"]` instead of `info["name"]`. I also decided to stick with the existing logic using the labels - in practice I see on my hub PowerOff is activityid `-1` but don't really want to introduce risk around that change right now.